harness/determined/cli/resource_pool.py

```diff
@@ -0,0 +1,166 @@
+from argparse import ONE_OR_MORE, Namespace
+from typing import Any, List
+
+from determined.cli import render, setup_session
+from determined.common.api import authentication, bindings
+from determined.common.declarative_argparse import Arg, Cmd
+
+
+def get_workspaces_string(workspaces: List[str]) -> str:
+    workspaces_str = ""
+    for workspace in workspaces[:-1]:
+        workspaces_str += workspace + ", "
+
+    if workspaces_str != "":
+        workspaces_str += "and "
+    workspaces_str += workspaces[-1]
+    return workspaces_str
+
+
+@authentication.required
+def add_binding(args: Namespace) -> None:
+    body = bindings.v1BindRPToWorkspaceRequest(
+        resourcePoolName=args.pool_name, workspaceNames=args.workspace_names
+    )
+    bindings.post_BindRPToWorkspace(setup_session(args), body=body, resourcePoolName=args.pool_name)
+
+    workspaces_string = get_workspaces_string(args.workspace_names)
+
+    print(
+        f'added bindings between the resource pool "{args.pool_name}" '
+        f"and the following workspaces: {workspaces_string}"
+    )
+    return
+
+
+@authentication.required
+def remove_binding(args: Namespace) -> None:
+    body = bindings.v1UnbindRPFromWorkspaceRequest(
+        resourcePoolName=args.pool_name,
+        workspaceNames=args.workspace_names,
+    )
+    bindings.delete_UnbindRPFromWorkspace(
+        setup_session(args), body=body, resourcePoolName=args.pool_name
+    )
+
+    workspaces_string = get_workspaces_string(args.workspace_names)
+    print(
+        f'removed bindings between the resource pool "{args.pool_name}'
+        f"and the following workspaces: {workspaces_string}"
+    )
+    return
+
+
+@authentication.required
+def replace_bindings(args: Namespace) -> None:
+    body = bindings.v1OverwriteRPWorkspaceBindingsRequest(
+        resourcePoolName=args.pool_name,
+        workspaceNames=args.workspace_names,
+    )
+    bindings.put_OverwriteRPWorkspaceBindings(
+        setup_session(args), body=body, resourcePoolName=args.pool_name
+    )
+
+    workspaces_string = get_workspaces_string(args.workspace_names)
+    print(
+        f'replaced bindings of the resource pool "{args.pool_name}'
+        f"with those to the following workspaces: {workspaces_string}"
+    )
+    return
+
+
+@authentication.required
+def list_workspaces(args: Namespace) -> None:
+    session = setup_session(args)
+    resp = bindings.get_ListWorkspacesBoundToRP(session, resourcePoolName=args.pool_name)
+    if resp.workspaceIds is None or len(resp.workspaceIds) == 0:
+        print("resource pool has no assignments")
+        return
+
+    print("ids are:", resp.workspaceIds)  # TODO: convert to names
+    workspace_names = ""
+    workspaces_set = set(resp.workspaceIds)
+    workspaces = bindings.get_GetWorkspaces(session)
+    for workspace in workspaces.workspaces:
+        if workspace.id in workspaces_set:
```

Comment:
This comment is super-nitpicky, and is something I'd generally not mention. But I've got the sense that you spend most of your time with other languages and their idioms so I'm erring on the side of "maybe more is helpful" rather than "maybe more is annoying"...

Genuinely don't know if it's more performant or not to construct a set of `workspaceIds` to test for inclusion, but I don't think it's worth the additional code either way. This whole loop probably feels most idiomatic as a list comprehension that's passed to a `join`. I think you could replace all of lines 81 - 89 with two (still readable) lines this way.

Reply:
thanks for the info! you're right...I definitely been coding in golang a lot more and this is definitely more go idiomatic 😅

Reply:
Nice. I like how this reads.

Stylistic nit: I think it's slightly cleaner without the conditional, but I don't know if that's a Pythonic thing or just me. It's good either way, we can consider it personal preference.